source/common/stream_info/filter_state_impl.cc

```diff
@@ -60,5 +79,54 @@ FilterState::Object* FilterStateImpl::getDataMutableGeneric(absl::string_view da
   return current->data_.get();
 }
 
+bool FilterStateImpl::hasDataAboveLifeSpan(FilterState::LifeSpan life_span) const {
```

Comment:
This is returning hasDataAtOrAboveLifeSpan, no?

Reply:
Correct. Renamed. I was trying to be brief, but apparently achieved confusion.